src/main/resources/productv2.avro

```diff
@@ -0,0 +1,55 @@
+{
+        "namespace": "datagen.example",
+        "name": "productv2",
+        "type": "record",
+        "fields": [
+                {"name": "id", "type": {
+                    "type": "long",
+                    "arg.properties": {
+                        "iteration": {
+                            "start": 0
+                        }
+                    }
+                }},
+                {"name": "name", "type": {
+                    "type": "string",
+                       "arg.properties": {
+                         "options": [
+                           "boots",
+                           "shirt",
+                           "knife",
+                           "saucepan",
+                           "table",
+                           "chair",
+                           "clock",
+                           "frame",
+                           "guitar",
+                           "thermometer",
+                           "scarf",
+                           "doormat",
+                           "vase",
+                           "clippers",
+                           "game",
+                           "towels"
+                         ]
+                       }
+                }},
+                {"name": "description", "type": {
+                    "type": "string",
+                    "arg.properties": {
+                      "iteration": {
+                          "start": 0
+                      }
+                    }
+                 }},
+                 {"name": "price", "type": {
```

Comment:
@C0urante just wanted to check in with you on this part.  `price` would ideally be of the dollar form as a `decimal` with 2 digits after the decimal point (unlike the `long` that it is now).  To that end, I tried a few different variations as demonstrated in https://github.com/confluentinc/avro-random-generator/blob/master/src/test/resources/test-schemas/decimals.json but they resulted in varying errors.  If you happen to have the cycles, would you be able to provide a working example of creating a realistic `price` field as a decimal?  

If we can't get that working, there is another option which is to construct it as a `string` like:

```
"regex":"[1-9][0-9]{0,1}[.][0-9]{2,2}"
```

Or we just leave it as it is...

Reply:
Hmm... would that be a decimal with a scale of 2? IIRC the way Avro represents decimal types is with an unscaled value multiplied by `10^(-1 * scale)`, so a scale of 2 would guarantee at most two digits after the decimal point.

Reply:
> a scale of 2 would guarantee at most two digits after the decimal point.

Yes, this is the desired outcome.  But all syntaxes that I tried resulted in errors.  For example:

```
[11:44:32] ~/git/kafka-connect-datagen/src/main/resources(retail) ✗: git diff
diff --git a/src/main/resources/productv2.avro b/src/main/resources/productv2.avro
index 002c579..98522ab 100644
--- a/src/main/resources/productv2.avro
+++ b/src/main/resources/productv2.avro
@@ -42,14 +42,14 @@
                       }
                     }
                  }},
-                 {"name": "price", "type": {
-                    "type": "long",
-                     "arg.properties": {
-                        "range": {
-                            "min": 2,
-                            "max": 20
-                        }
-                     }
-                  }}
+   {
+     "name": "price",
+     "type": {
+       "type": "bytes",
+       "logicalType": "decimal",
+       "precision": 4,
+       "scale": 2
+     }
+   }
         ]
 }
```

Results in:

```
[2021-09-13 11:44:36,828] ERROR Unable to parse the provided schema (io.confluent.kafka.connect.datagen.ConfigUtils:60)
org.apache.avro.SchemaParseException: com.fasterxml.jackson.core.io.JsonEOFException: Unexpected end-of-input: expected close marker for Object (start marker at [Source: (BufferedInputStream); line: 1, column: 1])
```

Reply:
Hmm... I was able to run `arg` (the Avro Random Generator CLI) locally against the latest `productv2` schema with this change:

```
-                    "type": "long",
-                     "arg.properties": {
-                        "range": {
-                            "min": 2,
-                            "max": 20
-                        }
-                     }
+                    "type": "bytes",
+                    "logicalType": "decimal",
+                    "scale": 2,
+                    "precision": 4
```

I can provide the complete schema if that'd be helpful, LMK.

Reply:
Ah, weird. Even though the `arg` CLI works, the connector tests fail with the `SchemaParseException` you provided.

Reply:
Okay, even weirder--seeing those same errors in the logs for builds on the latest for this branch, even without the changes to the `productV2` schema. They're not causing the build to fail, though.

Reply:
Looks like the "real" cause of the failure once the `productV2` schema is changed to use the logical decimal type is this:
```
java.lang.AssertionError
	at io.confluent.kafka.connect.datagen.DatagenTaskTest.assertRecordsMatchSchemas(DatagenTaskTest.java:270)
	at io.confluent.kafka.connect.datagen.DatagenTaskTest.generateAndValidateRecordsFor(DatagenTaskTest.java:235)
	at io.confluent.kafka.connect.datagen.DatagenTaskTest.shouldGenerateFilesForProductv2Quickstart(DatagenTaskTest.java:137)
```

Seems like the logic in `DatagenTaskTest::isConnectInstance` might need to be updated to handle `ByteBuffer` instances, based on [this arg logic](https://github.com/confluentinc/avro-random-generator/blob/f83f891c126aed54de892670cd7efbf6fb2c13a9/src/main/java/io/confluent/avro/random/generator/Generator.java#L1100-L1110).

Reply:
@C0urante can you please provide more guidance on the fix?  Why isn't [this case](https://github.com/confluentinc/kafka-connect-datagen/blob/master/src/test/java/io/confluent/kafka/connect/datagen/DatagenTaskTest.java#L257-L258) handling the logic?

Reply:
So I initially thought that that case wasn't sufficient because the Avro Random Generator produces `ByteBuffer` instances instead of `byte[]` instances when producing logical decimal data. However, it's actually slightly more complicated than that, since the connector itself then converts the Avro data produced by the generator to the Kafka Connect format using the `AvroData` class from the Avro converter. That conversion causes that `ByteBuffer` to be turned into a `BigDecimal` instance (see the [AvroData](https://github.com/confluentinc/schema-registry/blob/ff34ad373a3a59c9da812b93e04cecbd2646f1c7/avro-data/src/main/java/io/confluent/connect/avro/AvroData.java#L212-L214) and [Connect Decimal type](https://github.com/apache/kafka/blob/9628c1278e5dcd4f6995c89809461d350d9a530a/connect/api/src/main/java/org/apache/kafka/connect/data/Decimal.java#L73) logic).

So TL;DR: it's not that we have to update that case to handle `ByteBuffer` instances (although we probably should anyways), it's that we also have to add logic that accepts `BigDecimal` instances when the schema is the logical decimal type.

I have a commit that does this exact thing; I can push it to this branch if you're interested (and feel free to revert later if you'd like).

One thing I did notice when testing this is that it doesn't appear that range properties (i.e., min/max) are supported by the generator for logical decimal types. So although we could change price to a decimal type with a scale of two, we couldn't prevent it from having negative values, for example.

Reply:
@C0urante thank you for going down the rabbit hole.

> we couldn't prevent it from having negative values

This is a problem for the `price` field and is a deal breaker for this use case, so presumably we'll have to stick with the current implementation of this PR with price being a `long`.

IIUC the issues you highlighted could be fixed regardless.  Would it be possible for you to file an issue for tracking so we don't lose this idea?

Reply:
Sure, done (linked ticket to you offline).

Reply:
Got a PR up to add this to version 0.4.0 of the generator: https://github.com/confluentinc/avro-random-generator/pull/21

Reply:
PR's been merged, I think we can make this work with the new [0.4.0 release](https://github.com/confluentinc/avro-random-generator/tree/v0.4.0) of the generator.